### PR TITLE
Database: log how many rows DB result has

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/PLATFORM-836

Extends sampled logging of SQL queries introduced in #3787

@pchojnacki / @owend / @drsnyder

### CODE REVIEW

File: includes/db/Database.php [hunk 1]
```diff
@@ -3519,21 +3521,24 @@ protected function executeAndProfileQuery( $sql, $fname, $isMaster ) {
...
 		if ($this->getSampler()->shouldSample()) {
 			$this->getWikiaLogger()->info( "SQL $sql", [
 				'method'      => $fname,
 				'elapsed'     => $elapsedTime,
+				'num_rows'    => ( $ret instanceof ResultWrapper ? $ret->numRows() : mysql_num_rows( $ret ) ),
 				'cluster'     => $wgDBcluster,
 				'server'      => $this->mServer,
 				'server_role' => $isMaster ? 'master' : 'slave',
 				'db_name'     => $this->mDBname,
+				'exception'   => new Exception(), // log the backtrace
```
Comment (wladekb) [inline includes/db/Database.php]: we keep adding exception here and there, maybe we should add it by default in the logger?

Comment (macbre) [inline includes/db/Database.php]: I don't think backtrace is required in all cases (given the fact that it can be pretty big).

**Comment by wladekb** [comment]
lgtm

**Comment by drsnyder** [comment]
:+1:

**Pull request merged**